web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -108,19 +117,25 @@ export class LayerDialogComponent implements OnDestroy {
     if (!this.projectId) {
       throw Error('Project not yet loaded');
     }
+    let fields = Map<string, Field>();
+    this.layerForm.value.questions.map((question: { en: string }) => {
+      const fieldId = this.dataStoreService.generateId();
+      const field: Field = {
+        id: fieldId,
+        type: FieldType['TEXT'],
+        required: false,
+        label: StringMap(question),
+      };
+      fields = fields.set(fieldId, field);
```

Comment:
map != forEach; map is a functional operator whose lambda should return a value for each element.  Here it appears map() is being used as a proxy for a for loop. Please either convert to fully functional form, or replace with. forEach or for loop.